### PR TITLE
Streamlining the New Release Issue Template.

### PR DESCRIPTION
**Pull request by ryanwelcher**
## What?
This PR makes some small changes to the issue template based on using it for the last two releases of the Gutenberg plugin.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/New_release.md [hunk 1]
```diff
@@ -14,36 +14,34 @@ This issue is to provide visibility on the progress of the release process of Gu
...
 -   📽 Watch: [Gutenberg plugin v12.0.0 Release Party!](https://www.youtube.com/watch?v=4SDtpVPDsLc)
 
 ## Checklist
 
-### RC Day - Wednesday, March 15
+### RC Day - {Weekday, Month, Date}
 
 -   [ ] _Optional:_ Attend `#core-editor` meeting (14:00UTC)
 -   [ ] Post a message in `#core-editor` channel to let folks know you are starting the RC release process
 -   [ ] Organize and Label PRs on the relevant milestone
```
Comment (mburridge) [inline .github/ISSUE_TEMPLATE/New_release.md:27]: Should this be done before RC day? Or at least before posting in #core-editor that you're about to start the release process?

File: .github/ISSUE_TEMPLATE/New_release.md [hunk 2]
```diff
@@ -14,36 +14,34 @@ This issue is to provide visibility on the progress of the release process of Gu
...
 -   [ ] [Update the created Draft Release accordingly](https://developer.wordpress.org/block-editor/contributors/code/release/#view-the-release-draft)
 -   [ ] Publish Release
 -   [ ] Announce in `#core-editor` channel that RC1 has been released and is ready for testing
 -   [ ] Ping any other relevant channels announcing that the RC is available
--   [ ] Create Draft of Release post on Make Core blog _(initial draft in [Google doc](https://docs.google.com/document/d/1Hh25EYXSvRd5K45gq0VFN7yfN5l9om2FpX-_KqpVW7g/edit#))_
+-   [ ] Create Draft of Release post on Make Core blog _(initial draft in [Google doc](https://docs.google.com/document/d/1D-MTOCmL9eMlP9TDTXqlzuKVOg_ghCPm9_whHFViqMk/edit))_
 
 ### Between RC and Release
 
--   [ ] [Backport to RC](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22Backport+to+Gutenberg+RC%22+is%3Aclosed)
```
Comment (mburridge) [inline .github/ISSUE_TEMPLATE/New_release.md:38]: Why are you dropping this?

Comment (ryanwelcher) [reply .github/ISSUE_TEMPLATE/New_release.md:38]: It's being moved to the release day.

**Comment by bph** [comment]
Oh good, that you take another stab at it @ryanwelcher 
There is a typo in both place where you use workflow - the 'f' is missing. 
> Start the release process by triggering the rc [workflow](https://developer.wordpress.org/block-editor/contributors/code/release/#running-workflow)
and 
>  Start the release process by triggering the stable [workflow](https://developer.wordpress.org/block-editor/contributors/code/release/#running-workflow)

_I see that's already fixed. Ignore me_

File: .github/ISSUE_TEMPLATE/New_release.md (hunk 1 above)
Comment (ryanwelcher) [inline .github/ISSUE_TEMPLATE/New_release.md:27]: I'm not sure I understand. These are the steps that need to be taken during the RC release process.

File: .github/ISSUE_TEMPLATE/New_release.md [hunk 3]
```diff
@@ -14,36 +14,34 @@ This issue is to provide visibility on the progress of the release process of Gu
...
--   [ ] [Draft Release Post Highlights and Change Log](https://docs.google.com/document/d/1Hh25EYXSvRd5K45gq0VFN7yfN5l9om2FpX-_KqpVW7g/edit#)
+-   [ ] [Draft Release Post Highlights and Change Log](https://docs.google.com/document/d/1D-MTOCmL9eMlP9TDTXqlzuKVOg_ghCPm9_whHFViqMk/edit)
 -   [ ] Get assets from [Design Team](https://make.wordpress.org/design/) for the post
--   [ ] Reach out to Highlight Authors to draft sections _(if necessary)_
```
Comment (mburridge) [inline .github/ISSUE_TEMPLATE/New_release.md:41]: Not sure why you're dropping this too? Reaching out to authors of highlighted items to get them draft that part of the post could be useful, esp. if the item is particularly technical, or the release lead is less technical.

Comment (ryanwelcher) [reply .github/ISSUE_TEMPLATE/New_release.md:41]: We've never done this in any release that either myself or @justintadlock have done. It seemed extraneous.

File: .github/ISSUE_TEMPLATE/New_release.md (hunk 2 above)
Comment (mburridge) [reply .github/ISSUE_TEMPLATE/New_release.md:38]: Sorry, I commented before getting that far 🤦‍♂️

File: .github/ISSUE_TEMPLATE/New_release.md [hunk 4]
```diff
@@ -14,36 +14,34 @@ This issue is to provide visibility on the progress of the release process of Gu
 ## Resources
 
 -   📖 Read: [Gutenberg Release Process](https://developer.wordpress.org/block-editor/contributors/code/release/)
--   📖 Read: [Leading a Gutenberg Plugin Release](https://codep2.wordpress.com/2021/07/22/leading-a-gutenberg-plugin-release/)
```
Comment (bph) [inline .github/ISSUE_TEMPLATE/New_release.md:17]: This link should be removed as it's inaccessible for the community

Comment (ryanwelcher) [reply .github/ISSUE_TEMPLATE/New_release.md:17]: It has been removed :)

File: .github/ISSUE_TEMPLATE/New_release.md [hunk 5]
```diff
@@ -14,36 +14,34 @@ This issue is to provide visibility on the progress of the release process of Gu
...
 -   [ ] Announce in `#core-editor` channel that RC1 has been released and is ready for testing
 -   [ ] Ping any other relevant channels announcing that the RC is available
--   [ ] Create Draft of Release post on Make Core blog _(initial draft in [Google doc](https://docs.google.com/document/d/1Hh25EYXSvRd5K45gq0VFN7yfN5l9om2FpX-_KqpVW7g/edit#))_
+-   [ ] Create Draft of Release post on Make Core blog _(initial draft in [Google doc](https://docs.google.com/document/d/1D-MTOCmL9eMlP9TDTXqlzuKVOg_ghCPm9_whHFViqMk/edit))_
 
 ### Between RC and Release
 
--   [ ] [Backport to RC](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22Backport+to+Gutenberg+RC%22+is%3Aclosed)
--   [ ] [Draft Release Post Highlights and Change Log](https://docs.google.com/document/d/1Hh25EYXSvRd5K45gq0VFN7yfN5l9om2FpX-_KqpVW7g/edit#)
+-   [ ] [Draft Release Post Highlights and Change Log](https://docs.google.com/document/d/1D-MTOCmL9eMlP9TDTXqlzuKVOg_ghCPm9_whHFViqMk/edit)
```
Comment (bph) [inline .github/ISSUE_TEMPLATE/New_release.md:37]: Thanks for making it a reusable template :-)

**Review by mburridge — APPROVED** [review]

**Pull request merged**